packages/ui/src/views/docstore/AddDocStoreDialog.jsx

```diff
@@ -0,0 +1,249 @@
+        else dispatch({ type: HIDE_CANVAS_DIALOG })
+        return () => dispatch({ type: HIDE_CANVAS_DIALOG })
+    }, [show, dispatch])
+
+    const createDocumentStore = async () => {
+        try {
+            const obj = {
+                name: documentStoreName,
+                description: documentStoreDesc
+            }
+            const createResp = await documentStoreApi.createDocumentStore(obj)
+            if (createResp.data) {
+                enqueueSnackbar({
+                    message: 'New Document Store created.',
+                    options: {
+                        key: new Date().getTime() + Math.random(),
+                        variant: 'success',
+                        action: (key) => (
+                            <Button style={{ color: 'white' }} onClick={() => closeSnackbar(key)}>
+                                <IconX />
+                            </Button>
+                        )
+                    }
+                })
+                onConfirm(createResp.data.id)
+            }
+        } catch (err) {
+            const errorData = typeof err === 'string' ? err : err.response?.data || `${err.response?.status}: ${err.response?.statusText}`
+            enqueueSnackbar({
+                message: `Failed to add new Document Store: ${errorData}`,
+                options: {
+                    key: new Date().getTime() + Math.random(),
+                    variant: 'error',
+                    persist: true,
+                    action: (key) => (
+                        <Button style={{ color: 'white' }} onClick={() => closeSnackbar(key)}>
+                            <IconX />
+                        </Button>
+                    )
+                }
+            })
+            onCancel()
+        }
+    }
+
+    const updateDocumentStore = async () => {
+        try {
+            const saveObj = {
+                name: documentStoreName,
+                description: documentStoreDesc
+            }
+
+            const saveResp = await documentStoreApi.updateDocumentStore(docStoreId, saveObj)
+            if (saveResp.data) {
+                enqueueSnackbar({
+                    message: 'Document Store Updated!',
+                    options: {
+                        key: new Date().getTime() + Math.random(),
+                        variant: 'success',
+                        action: (key) => (
+                            <Button style={{ color: 'white' }} onClick={() => closeSnackbar(key)}>
+                                <IconX />
+                            </Button>
+                        )
+                    }
+                })
+                onConfirm(saveResp.data.id)
+            }
+        } catch (error) {
+            const errorData = error.response?.data || `${error.response?.status}: ${error.response?.statusText}`
+            enqueueSnackbar({
+                message: `Failed to update Document Store: ${errorData}`,
+                options: {
```

Comment:
`message: `Failed to add new Document Store: ${error.response.data.message}`,`

Reply:
fixed.